backend/CircleForms/Database/Services/PostRepository.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using CircleForms.Database.Models.Posts;
+using CircleForms.Database.Models.Users;
+using CircleForms.Database.Services.Abstract;
+using Microsoft.Extensions.Logging;
+using MongoDB.Driver;
+using MongoDB.Entities;
+
+namespace CircleForms.Database.Services;
+
+public class PostRepository : IPostRepository
+{
+    private readonly ILogger<PostRepository> _logger;
+
+    public PostRepository(ILogger<PostRepository> logger)
+    {
+        _logger = logger;
+    }
+
+    public async Task<Post> Add(string userId, Post post)
+    {
+        post.AuthorRelation = userId;
+        post.PublishTime = DateTime.UtcNow;
+
+        _logger.LogInformation("User {Id} created a new post", userId);
+        _logger.LogDebug("User {Id} created a new post {@Post}", userId, post);
+
+        await post.SaveAsync();
+
+        await DB.Entity<User>(userId).PostsRelation.AddAsync(post);
+
+        return post;
+    }
+
+    public async Task<List<Post>> Get()
+    {
+        return await DB.Fluent<Post>().ToListAsync();
+    }
+
+    public async Task<Post> Get(string postId)
+    {
+        return await DB.Find<Post>().OneAsync(postId);
+    }
+
+    public async Task AddAnswer(Post post, Answer entry)
+    {
+        //TODO: transactions

```

Comment:
Considering its a single operation, don't think we even need a transaction here. It shouldn't be able to fail either